### PR TITLE
Add some guidelines for refactoring

### PR DESCRIPTION
**Pull request by wxiaoguang**
Just some brief ideas.

Feel free to complete these guidelines, feel free to edit on this PR directly.

### CODE REVIEW

File: docs/content/doc/developers/guidelines-refactoring.en-us.md
Comment (wolfogre) [inline docs/content/doc/developers/guidelines-refactoring.en-us.md:49]: So it's not just a suggestion, it also introduces a new rule, as a patch of https://github.com/go-gitea/gitea/blob/49919c636e4788a14f3762a0d9137ee2e0092790/CONTRIBUTING.md?plain=1#L122-L125

So we have to update `CONTRIBUTING.md`.

I would agree with:

1. If a PR has been ignored (no comment, no review) for a long time (more than 7 days), and the author or a maintainer believe it won't stand a long wait (like a refactoring PR), he/she can send a last call to TOC or all maintainers.
2. After another 7 days, if there is still zero approval, that means a polite refusal, then the PR will be closed to avoid wasting more time, so "a last call" has a cost, use it carefully.
3. However, if there is no objection from TOC, the PR could be merged with one core member's approval (not the author).

Comment (wxiaoguang) [reply docs/content/doc/developers/guidelines-refactoring.en-us.md:49]: TOC decides. Feel free to edit this PR directly.

Comment (zeripath) [reply docs/content/doc/developers/guidelines-refactoring.en-us.md:49]: I think having a call out to TOC would be very useful - I think we've just started doing this in any case.

**Review by delvh — APPROVED** [review]

**Review by lunny — APPROVED** [review]

File: docs/content/doc/developers/guidelines-refactoring.en-us.md
Comment (wxiaoguang) [inline docs/content/doc/developers/guidelines-refactoring.en-us.md:30]:
```suggestion
```

**Pull request merged**
